### PR TITLE
Implemented Complex Resolving feature

### PR DESCRIPTION
**Pull request by vilchev**
issue: https://github.com/eclipse/smarthome/issues/1247

Signed-off-by: Vasil Ilchev <v.ilchev@prosyst.bg>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/ReferenceResolverUtil.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+ * <br/>
+ * 'childInput' : '$compositeModuleInput'
+ * </li>
+ * <li>
+ * Single reference to CompositeModule configuration where whole input value is replaces with the referenced value
+ * <br/>
+ * 'childInput' : '$compositeModuleConfiguration'
+ * </li>
+ * </ul>
+ */
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/ReferenceResolverUtil.java]: Could you please add an "author" tag here?

**Comment by kaikreuzer** [comment]
lgtm (but haven't actively tested it).
Might it be possible to add a dedicated test for this functionality? I seems possible to add an integration test that checks the overall feature as well as having unit tests for the methods of `ReferenceResolverUtil`.

**Comment by kaikreuzer** [comment]
Thanks!

**Comment by maggu2810** [comment]
@vilchev It seems this PR breaks the Automation Integration Test -- at least on two of my machines.
Does it work for you?

**Pull request merged**